dockerfiles/init/manifests/codenvy.env

```diff
@@ -86,17 +86,121 @@
 #CODENVY_SERVER_XMX=2048
 
 
-
 ########################################################################################
 #####                                                                              #####
-#####                                   DOCKER                                     #####
+#####                                 NETWORKING                                   #####
 #####                                                                              #####
-# IP Address
-#     The IP address of the Docker daemon that is running on your host. We do a 
-#     self-discovery to set this. You can combine this with values of DOCKER_HOST to 
-#     change communications from socket to TCP.
+#
+# Affects how the Codenvy server connects to the Docker Daemon, how the Che server and
+# workspaces establish connections to one another, and also how remote browser clients
+# should discover and connect to workspaces.
+#
+# Codenvy goes through a progression algorithm to establish the protocol, IP address and
+# port to establish communications with it is booting or starting a workspace.
+#
+# Browser --> Codenvy Server
+#    1. Default is 'http://localhost:${SERVER_PORT}/wsmaster/api'.
+#    2. Else use the value of CODENVY_API
+#
+# Codenvy Server --> Docker Daemon Progression:
+#    1. Use the value of CODENVY_DOCKER_DAEMON__URL
+#    2. Else, use the value of DOCKER_HOST system variable
+#    3. Else, use Unix socket over unix:///var/run/docker.sock
+#
+# Codenvy Server --> Workspace Connection:
+#    - If CODENVY_DOCKER_SERVER__EVALUATION__STRATEGY is 'default':
```

Comment:
In the case of Che, users / admins can configure certain variables because it's not always the case that they can get Che -> Workspace to connect.  So they have to configure variables to make this connection work.

If in Codenvy there is *nothing* that an end user admin will need to configure and that we guarantee that Codenvy -> Worskpace can always make a connection, then we do not have to disclose the strategy and can just call it internal.  But if there are any variables that affect how Codenvy discovers a ws, then I need to document it.